services/src/util/server.rs

```diff
@@ -294,3 +300,47 @@ pub(crate) fn log_server_info() -> Result<()> {
 pub async fn not_implemented_handler() -> HttpResponse {
     HttpResponse::NotImplemented().finish()
 }
+
+pub struct SocketFd(pub RawFd);
+
+// attach the connections socket file descriptor to the connection data
+// TODO: the socket file descriptor might be re-used by a subsequent connection before we notice that
+//       the connections was closed. We maybe need a global list of all open connections and their socket file descriptors
+//       or some other way to invalidate the socket file descriptor in the old connection when a new one arrives.
+pub fn connection_init(connection: &dyn Any, data: &mut Extensions) {
```

Comment:
we should add a test here

Reply:
what would you want to test? There is no logic happening here.